flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/FrontMetricGroup.java

```diff
@@ -72,8 +74,15 @@ public MetricGroup getWrappedMetricGroup() {
 
     @Override
     public Map<String, String> getAllVariables() {
-        return parentMetricGroup.getAllVariables(
-                this.settings.getReporterIndex(), this.settings.getExcludedVariables());
+        Map<String, String> allVariables =
+                parentMetricGroup.getAllVariables(
+                        this.settings.getReporterIndex(), this.settings.getExcludedVariables());
+
+        if (this.settings.getAdditionalVariables().isEmpty()) {
+            return allVariables;
+        }
+
+        return new CompositeMap(allVariables, this.settings.getAdditionalVariables());
```

Comment:
@zentol Thanks for the `CompositeMap` suggestion. I didn't add a mutator for collisions here--I figure throwing an exception is acceptable since we wouldn't want users overriding Flink internal metric variables.

Reply:
I think that's fine, but we could also wrap it in `Collections.unmodifiableMap()`, because we don't want any modifications _at all_.

Reply:
Which admittedly is technically a separate change, but we are changing the behavior anyway.